src/app/pages/PerpetualPage/components/TradeForm/index.tsx

```diff
@@ -44,18 +47,41 @@ export const TradeForm: React.FC<ITradeFormProps> = ({
   const { t } = useTranslation();
   const { checkMaintenance, States } = useMaintenance();
   const inMaintenance = checkMaintenance(States.PERPETUAL_TRADES);
+  const ammState = usePerpetual_queryAmmState();
+  const perpParameters = usePerpetual_queryPerpParameters();
+
+  const [lotSize, lotPrecision] = useMemo(() => {
+    const lotSize = Number(perpParameters.fLotSizeBC.toPrecision(8));
+    const lotPrecision = lotSize.toString().replace(/^.*\.?/, '').length;
+    return [lotSize, lotPrecision];
+  }, [perpParameters.fLotSizeBC]);
+
+  const maxTradeSize = useMemo(
+    () =>
+      Number(
+        Math.abs(
+          getMaximalTradeSizeInPerpetual(
+            0,
```

Comment:
This will need to be adjusted but I can do it in https://github.com/DistributedCollective/Sovryn-frontend/pull/1856 .